src/main/java/fi/aalto/cs/apluscourses/ui/ideactivities/ComponentDatabase.java

```diff
@@ -101,6 +116,69 @@ public static boolean showToolWindow(@NotNull String id, @NotNull Project projec
     return false;
   }
 
+  /**
+   * Opens the Run tool window and starts the REPL for a given module.
+   */
+  public static boolean startAndShowRepl(@NotNull Module module, @NotNull Project project) {
+    var toolWindow = ToolWindowManager.getInstance(project).getToolWindow(RUN_TOOL_WINDOW);
+    if (toolWindow == null || !showToolWindow(RUN_TOOL_WINDOW, project)) {
+      return false;
+    }
+
+    if (Arrays.stream(toolWindow.getContentManager().getContents()).anyMatch(content ->
+        content.toString().contains("REPL") && content.toString().contains(module.getName()))) {
+      return true;
+    }
+
+    var runConfigs = RunManager.getInstance(project).getAllConfigurationsList();
+    for (var config : runConfigs) {
+      var name = config.getName();
+      if (name.contains("REPL") && name.contains(module.getName())) {
+
+        var settings = RunManager.getInstance(project).findSettings(config);
+        var processHandler = ScalaConsoleInfo.getProcessHandler(project);
+        if (settings != null && processHandler != null) {
+          ExecutionManager.getInstance(project).restartRunProfile(project,
+              DefaultRunExecutor.getRunExecutorInstance(),
+              DefaultExecutionTarget.INSTANCE,
+              settings,
+              processHandler);
+          return true;
+        }
+      }
+    }
+    return false;
+  }
+
+  /**
+   * Closes all REPLs for a project.
+   */
+  public static void closeRepls(@NotNull Project project) {
+    var descriptors = ExecutionManagerImpl.getAllDescriptors(project);
+    for (var descriptor : descriptors) {
+      var processHandler = descriptor.getProcessHandler();
+      if (processHandler != null && descriptor.getDisplayName().contains("REPL")) {
+        processHandler.destroyProcess();
+        var started = new Date();
+        while (!processHandler.isProcessTerminated()) {
+          var now = new Date();
+          if (now.getTime() - started.getTime() > 500) {
+            break;
+          }
+        }
```

Comment:
I'd prefer Thread.sleep rather than spinning in a loop eating 100% of the CPU, but I guess it's such a short operation that it doesn't really matter